src/services/polkit/flow.hpp

```diff
@@ -0,0 +1,162 @@
+#pragma once
+
+#include <qobject.h>
+#include <qqmlintegration.h>
+
+#include "../../core/retainable.hpp"
+#include "identity.hpp"
+#include "listener.hpp"
+
+namespace qs::service::polkit {
+class Session;
+
+class AuthFlow
+    : public QObject
+    , public Retainable {
+	Q_OBJECT;
+	QML_ELEMENT;
+	QML_UNCREATABLE("AuthFlow can only be obtained from PolkitAgent.");
+
+	// clang-format off
+	/// The main message to present to the user.
+	Q_PROPERTY(QString message READ message CONSTANT);
+
+	/// The icon to present to the user in association with the message.
+	///
+	/// The icon name follows the [FreeDesktop icon naming specification](https://specifications.freedesktop.org/icon-naming-spec/icon-naming-spec-latest.html).
+	Q_PROPERTY(QString iconName READ iconName CONSTANT);
+
+	/// The action ID represents the action that is being authorized.
+	///
+	/// This is a machine-readable identifier.
+	Q_PROPERTY(QString actionId READ actionId CONSTANT);
+
+	/// A cookie that identifies this authentication request.
+	///
+	/// This is an internal identifier and not recommended to show to users.
+	Q_PROPERTY(QString cookie READ cookie CONSTANT);
+
+	/// The list of identities that may be used to authenticate.
+	///
+	/// Each identity may be a user or a group. You may select any of them to
+	/// authenticate by setting @@selectedIdentity. By default, the first identity
+	/// in the list is selected.
+	Q_PROPERTY(QList<Identity*> identities READ identities CONSTANT);
+
+	/// The identity that will be used to authenticate.
+	/// 
+	/// Setting this will abort any ongoing authentication conversations and start a new one.
+	Q_PROPERTY(Identity* selectedIdentity READ selectedIdentity WRITE setSelectedIdentity NOTIFY selectedIdentityChanged);
```

Comment:
"Setting" -> "Changing"

Reactive changes can cause a lot of "Setting" so this wording could be confusing